lib/install/validate/validator.go

```diff
@@ -482,3 +512,68 @@ func (v *Validator) AddDeprecatedFields(ctx context.Context, conf *config.Virtua
 
 	return &dconfig
 }
+
+// helper to grab HostSystem references out of the Session cluster
+func (v *Validator) getHostsInCluster(ctx context.Context) ([]*object.HostSystem, error) {
```

Comment:
Didn't see much value to have this method. 
What you want is actually to find hosts in cluster. if the result of Cluster.Hosts is empty, the later test will find that. 
And you need to check error in the caller no matter how, so suggest to remove this method.